omod/src/main/webapp/resources/dhisconnector-runreports.js

```diff
@@ -160,6 +179,26 @@ function handleYearlyPeriodChange() {
     selectedEndDate = moment(selectedYear, 'YYYY').endOf('year').toDate();
 }
 
+function handleFinancialYearPeriodChange(month) {
+    const yearlyPicker = jQuery('#financialJulyPicker');
+    let selectedYear = yearlyPicker.val();
+    const currentYear = moment().year();
+    const currentMonth = moment().month();
+    // Set back to the maximum possible year if the user entered a wrong value
+    if (selectedYear >= currentYear) {
+        if (currentMonth > moment().month(month).format("M")) {
+            yearlyPicker.val(currentYear - 1);
+            selectedYear = currentYear - 1;
+        } else {
+            yearlyPicker.val(currentYear - 2);
+            selectedYear = currentYear - 2;
+        }
+    }
+    selectedPeriod = selectedYear + month;
+    selectedStartDate = moment(selectedYear, 'YYYY').toDate();
+    selectedEndDate = moment(selectedYear, 'YYYY').endOf('year').toDate();
```

Comment:
This calculation is wrong. 

If it is financial july 2019, the start date should be 01 july 2019 and the end date should be 30th June 2020